src/main/java/TicTacToe/logic/Minimax.java

```diff
@@ -1,4 +1,7 @@
-package TicTacToe;
+package TicTacToe.logic;
+
+import TicTacToe.Game;
+import TicTacToe.board.Tile;
```

Comment:
W tej klasie, w metodzie `minimax` jest wywołanie rekurencyjne. Czyli z `minimax` wywołujesz `minimax`. To jest na 99% przyczyna wolnego działania gry - im więcej jest wolnych pól, tym dłużej trwa wyliczenie. Jesteś w stanie wytłumaczyć ten algorytm, dlaczego akurat tak to działa?
Generalnie rekurencja nie jest tutaj potrzebna, i tak nie jesteś w stanie przewidzieć co zrobi przeciwnik kilka ruchów dalej.
Można to zrobić prosto, np.
- w pierwszym kroku upewnić się, że blokujemy wygraną przeciwnika w kolejnej turze
- w drugim kroku, dostawić do już postawionego kółka tak, aby w kolejnym ruchu móc mieć 3